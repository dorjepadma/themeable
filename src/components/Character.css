.Character {
  height: 60%;
  background-color: rgb(160, 160, 221);
  color: black;
  border: rgb(12, 2, 59);
  margin: auto;
  }
  img {
    display: block;
    margin: auto;
    width: 60%;
    height:60% 
  }
  h2 {
    text-align: center;
  }
  
