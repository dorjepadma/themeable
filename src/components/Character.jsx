import React from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

const Character = ({ photoUrl, name, }) => (
  <section className={styles.Character}>
    <img src={photoUrl} 
      alt={name} />
    <h2>{name}</h2>
  </section>
);

Character.propTypes = {
  photoUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string
};

export default Character;

