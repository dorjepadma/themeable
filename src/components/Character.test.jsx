import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('creates the character', () => {
  it('builds the Character', () => {
    const wrapper = shallow (<Character
      name='Aang'
      photoUrl='https://vignette.wikia.nocookie.net/avatar/images/a/ae/Aang_at_Jasmine_Dragon.png/revision/latest?cb=20130612174003' />);
    expect(wrapper).toMatchSnapshot();
  });
});

