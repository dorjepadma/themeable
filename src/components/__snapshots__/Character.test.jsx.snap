// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates the character builds the Character 1`] = `
<section
  className="Character"
>
  <img
    alt="Aang"
    src="https://vignette.wikia.nocookie.net/avatar/images/a/ae/Aang_at_Jasmine_Dragon.png/revision/latest?cb=20130612174003"
  />
  <h2>
    Aang
  </h2>
</section>
`;
